scripts/gitlab/test_linux_stable.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/sh
+
+
+github_api_polkadot_pull_url="https://api.github.com/repos/paritytech/polkadot/pulls"
+# use github api v3 in order to access the data without authentication
+github_header="Authorization: token ${GITHUB_PR_TOKEN}"
+
+boldprint () { printf "|\n| \033[1m${@}\033[0m\n|\n" ; }
+boldcat () { printf "|\n"; while read l; do printf "| \033[1m${l}\033[0m\n"; done; printf "|\n" ; }
+
+
+
+POLKADOT_PATH=$(pwd)
+
+
+prepare_git(){
+  # Set the user name and email to make merging work
+  git config --global user.name 'CI system'
+  git config --global user.email '<>'
+}
+
+
+prepare_substrate(){
+  pr_companion=$1
+  boldprint "companion pr specified/detected: #${pr_companion}"
+
+  # Clone the current Substrate master branch into ./substrate.
+  # NOTE: we need to pull enough commits to be able to find a common
+  # ancestor for successfully performing merges below.
+  git clone --depth 20 https://github.com/paritytech/substrate.git
+  cd substrate
+  SUBSTRATE_PATH=$(pwd)
+
+  git fetch origin refs/pull/${pr_companion}/head:pr/${pr_companion}
+  git checkout pr/${pr_companion}
+  git merge origin/master
+
+  cd "$POLKADOT_PATH"
+
+  # Merge master into our branch before building Polkadot to make sure we don't miss
+  # any commits that are required by Polkadot.
+  git merge origin/master
```

Comment:
Yes I don't like these `git merge` either. My reason is not because it's fallible but because it is testing something different than the actual code in the branch. But again, I just copied that from the original script, I did not intend to put that in question.

Reply:
ok, should be resolved elsewhere

Reply:
@gabreal do you know if those merge are absolutely required? They can make a PR fails for no good reason: https://gitlab.parity.io/parity/polkadot/-/jobs/596036